src/main/scala/services/ValidationService.scala

```diff
@@ -17,55 +17,51 @@ object ValidationService {
 
   type ValidationResult[A] = ValidatedNec[ValidationError, A]
 
-  private def toGender(gender: Char): Either[ValidationError, Gender] = gender match {
-    case 'M' => Either.right(Male)
-    case 'F' => Either.right(Female)
-    case _ => Either.left(InvalidGenderError)
-  }
+  private def logErrors[F[_]: Sync](logger: Logger)(errors: NonEmptyChain[ValidationError]): F[Unit] = Sync[F].delay(
+    errors.toList.foreach(e =>
```

Comment:
In your code, you're iterating over a list and creating a log statement per item, but the "full list" (is not a real list since you're using `foreach`) is inside of a `delay` block. We want to have one `delay` per log statement.